diff --git a/field_collection.module b/field_collection.module
index b62d254..25cf0cc 100644
--- a/field_collection.module
+++ b/field_collection.module
@@ -302,7 +302,9 @@ function field_collection_item_access($op, FieldCollectionItemEntity $item = NUL
  */
 function field_collection_item_delete($id) {
   $fci = field_collection_item_load($id);
-  $fci->delete();
+  if ($fci) {
+    $fci->delete();
+  }
 }
 
 /**
