diff --git a/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module b/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module
index 67242cc..cf9a845 100644
--- a/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module
+++ b/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module
@@ -23,6 +23,17 @@ function ds_bootstrap_layouts_ds_layout_info() {
     'image' => TRUE,
   );
 
+  $layouts['bootstrap_12_stacked'] = array(
+    'label' => t('Bootstrap One Column: 12 stacked'),
+    'path' => $onecol . '/bootstrap_12_stacked',
+    'regions' => array(
+      'top' => t('Top'),  
+      'central' => t('Content'),
+    ),
+    'form' => TRUE,
+    'image' => TRUE,
+  );
+
   // Two columns.
   $layouts['bootstrap_1_11'] = array(
     'label' => t('Bootstrap Two Columns: 1-11'),
