diff --git a/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module b/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module
index d355843..67242cc 100644
--- a/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module
+++ b/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/ds_bootstrap_layouts.module
@@ -239,6 +239,7 @@ function ds_bootstrap_layouts_ds_layout_info() {
     'label' => t('Bootstrap Three Columns: 4-5-3'),
     'path' => $threecol . '/bootstrap_4_5_3',
     'regions' => array(
+      'top' => t('Top'),    
       'left' => t('Left'),
       'central' => t('Middle'),
       'right' => t('Right'),
diff --git a/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/layouts/threecol/bootstrap_4_5_3/bootstrap-4-5-3.tpl.php b/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/layouts/threecol/bootstrap_4_5_3/bootstrap-4-5-3.tpl.php
index 167d42e..2cb0b41 100644
--- a/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/layouts/threecol/bootstrap_4_5_3/bootstrap-4-5-3.tpl.php
+++ b/profiles/rouergue/modules/contrib/ds_bootstrap_layouts/layouts/threecol/bootstrap_4_5_3/bootstrap-4-5-3.tpl.php
@@ -11,6 +11,10 @@
     <?php print render($title_suffix['contextual_links']); ?>
   <?php endif; ?>
   <div class="row">
+          <<?php print $top_wrapper; ?> class="col-sm-12 <?php print $top_classes; ?>">
+        <?php print $top; ?>
+      </<?php print $top_wrapper; ?>>
+      
     <<?php print $left_wrapper; ?> class="col-sm-4 <?php print $left_classes; ?>">
       <?php print $left; ?>
     </<?php print $left_wrapper; ?>>
